TST: test pollution in scipy-only tests

As I mentioned in https://github.com/astropy/astropy/pull/17732#pullrequestreview-2602008107, I'm currently seeing 12 tests consistently fail when running double tests locally[^1]

```
❯ pytest --color=no --keep-duplicates --pyargs astropy astropy > double_test.log 2>&1
```

```
...
=========================== short test summary info ============================
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_1[model0]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_1[model1]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_1[model2]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_1[model3]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_1[model4]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_2[model0]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_2[model1]
FAILED astropy/modeling/tests/test_input.py::test_call_keyword_args_2[model2]
FAILED astropy/units/tests/test_quantity_ufuncs.py::TestUfuncHelpers::test_scipy_registered
FAILED astropy/units/tests/test_quantity_ufuncs.py::test_scipy_registration
FAILED astropy/visualization/wcsaxes/tests/test_wcsapi.py::test_coord_type_from_ctype
FAILED astropy/visualization/wcsaxes/tests/test_wcsapi.py::test_custom_coord_type_1d_2d_wcs_overwrite
==== 12 failed, 57587 passed, 752 skipped, 446 xfailed in 302.79s (0:05:02) ====
```

I checked that this reproduced even from a fresh clone + fresh env (resolved via `uv pip install -e [test_all]`), so I'm confident it's not a side effect of my own copy being dirty. For reference it fails the same way with numpy 1.26 or numpy 2.
I also tried running the same invocation with `v6.1.x`[^2] (branch) and saw no failures there.
However `v7.0.0` (tag) shows[^3] the same 12 failures.

I intend to try out https://pypi.org/project/detect-test-pollution/ to narrow it down and will report back when I know more about this issue.

[^1]: this was already the case with the previous invocation,`python -c 'import sys; from astropy import test; test(); sys.exit(test())'`, so #17732 is not the culprit.
[^2]: specifically with Python 3.12 + `numpy<2`, otherwise some of `units` completeness test don't pass, which is basically noise in this context
[^3]: specifically with Python 3.13 + `numpy<2.2`, for the same reason as 2.

--- a/docs/io/ascii/read.rst
+++ b/docs/io/ascii/read.rst
@@ -793,9 +793,11 @@ the :ref:`io_registry` (note that our format has no mechanism to write out the u
 .. testcleanup::
 
     >>> from astropy.io import registry
+    >>> from astropy.io.ascii.core import FORMAT_CLASSES
     >>> for format_name in ['custom_no_header', 'custom_commented_header', 'fixed_width_commented_header']:
     ...     registry.unregister_reader(f"ascii.{format_name}", Table)
     ...     registry.unregister_writer(f"ascii.{format_name}", Table)
+    ...     del FORMAT_CLASSES[format_name]
 
 **Define a custom reader functionally**
 

--- a/docs/time/index.rst
+++ b/docs/time/index.rst
@@ -1645,6 +1645,11 @@ standard `~astropy.time.TimeISO` class from which it inherits::
   >>> t2.iso
   '2016-01-01 00:00:00.000'
 
+.. testcleanup::
+
+  >>> from astropy.time import TIME_FORMATS
+  >>> del TIME_FORMATS["yday_custom"]
+
 .. EXAMPLE END
 
 .. EXAMPLE START: Customizing the TimeFormat Class with Time Since an Epoch
@@ -1674,6 +1679,11 @@ from the `~astropy.time.TimeFromEpoch` class and define a few class attributes::
   np.float64(946684832.0)
   >>> t.unix_leap - t.unix
   np.float64(32.0)
+
+.. testcleanup::
+
+  >>> from astropy.time import TIME_FORMATS
+  >>> del TIME_FORMATS["unix_leap"]
 
 .. EXAMPLE END
 

--- a/docs/units/combining_and_defining.rst
+++ b/docs/units/combining_and_defining.rst
@@ -116,4 +116,8 @@ Now, the prefixed units can be parsed etc::
   >>> print(u.Unit("megafoo").to(u.Unit("kFo")))
   1000.0
 
+.. testcleanup::
+
+    >>> u.core._unit_registries.pop()  # doctest: +IGNORE_OUTPUT
+
 .. EXAMPLE END

--- a/docs/units/standard_units.rst
+++ b/docs/units/standard_units.rst
@@ -262,6 +262,9 @@ To enable Imperial units, do::
       yd           | 0.9144 m        | yard                             ,
     ]
 
+.. testcleanup::
+
+    >>> u.core._unit_registries.pop()  # doctest: +IGNORE_OUTPUT
 
 This may also be used with the `Python "with" statement
 <https://docs.python.org/3/reference/compound_stmts.html#with>`_, to

--- a/docs/utils/iers.rst
+++ b/docs/utils/iers.rst
@@ -269,3 +269,7 @@ UT1Flag, PM_x, PM_y, PolPMFlag::
   57877.0 -0.6573705        P 0.010328 0.451777         P
   57878.0 -0.6587712        P 0.011924 0.453209         P
   57879.0  -0.660187        P 0.013544 0.454617         P
+
+.. testcleanup::
+
+   >>> iers.earth_orientation_table.set(None)  # doctest: +IGNORE_OUTPUT

--- a/tox.ini
+++ b/tox.ini
@@ -108,7 +108,7 @@ commands =
     !cov-!double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} {posargs}
     cov-!double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} --cov astropy --cov-config={toxinidir}/pyproject.toml --cov-report xml:{toxinidir}/coverage.xml {posargs}
 
-    double: pytest --keep-duplicates --pyargs astropy astropy
+    double: pytest --keep-duplicates --pyargs astropy {toxinidir}/docs astropy {toxinidir}/docs {env:MPLFLAGS} {posargs}
 
 pip_pre =
     devdeps: true

